connectors/slack/template/index.js

```diff
@@ -1,24 +1,321 @@
+const Url = require('url');
 const configure = require('./configure');
 const Sdk = require('@fusebit/add-on-sdk');
+const { verifyRequestSignature } = require('@slack/events-api');
+const storage = require('./storage');
+const Superagent = require('superagent');
 
-/*
+/**
+ * @param ctx {FusebitContext}
+ */
+module.exports = (ctx, cb) => {
+    let url = Url.parse(ctx.url);
+    let pathname = url.pathname.split('/');
+    pathname.splice(0, 5);
+    pathname = '/' + pathname.join('/');
+    url.pathname = pathname;
+    url = Url.format(url);
 
-Add-on Settings Manager contract: 
+    Sdk.debug('NEW REQUEST', ctx.method, ctx.url);
 
-Initialization - GET with query params
-- [required] returnTo (returnTo, required)
-- [optional] state (opaque, to roundtrip)
-- [optional] data (base64 encoded JSON object, to augment and roundtrip)
+    // Most requests go to the Settings Manager
+    if (!pathname.startsWith('/events')) {
+        Sdk.createSettingsManager(configure)(ctx).then(
+            function (result) {
+                cb(null, result);
+            },
+            function (err) {
+                cb(err, null);
+            }
+        );
+    } else {
+        if (ctx.configuration.slack_signing_secret) {
+            if (pathname.startsWith('/events/ingest') && ctx.method === 'POST') {
+                handleEventIngest(ctx, cb);
+            } else if (pathname.startsWith('/events/registration') && ctx.method === 'POST') {
+                return registerCore(ctx, cb);
+            } else if (pathname.startsWith('/events/registration') && ctx.method === 'DELETE') {
+                return registerCore(ctx, cb, true);
+                // COMMENT OUT WHEN NOT DEBUGGING
```

Comment:
Probably worth deleting, then, before publishing.

Reply:
This is super useful, I use it almost always, so I prefer to leave it in